core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -400,8 +400,15 @@ public void replaceTableViaTransactionThatAlreadyExistsAsView() {
                     .buildTable(viewIdentifier, SCHEMA)
                     .replaceTransaction()
                     .commitTransaction())
-        .isInstanceOf(NoSuchTableException.class)
-        .hasMessageStartingWith("Table does not exist: ns.view");
+        .satisfiesAnyOf(
+            throwable ->
+                assertThat(throwable)
+                    .isInstanceOf(NoSuchTableException.class)
+                    .hasMessageStartingWith("Table does not exist: ns.view"),
+            throwable ->
+                assertThat(throwable)
```

Comment:
Nessie throws `AlreadyExistsException` for a view in this case. Do we need to wrap it up with `NoSuchTableException` for a table in this case to generalize like other catalogs?

Reply:
Even Hive view support has same problem
https://github.com/apache/iceberg/pull/8907

Reply:
Altering code test expectation on a PR that adds a specific view implementation does not look right to me.

If core test expectations are sub-optimal perhaps we can adjust them in a separate PR, the rebase the Nessie view PR.

That said, I think the exception thrown in this test case should be the same across all catalog implementations. WDYT?

Reply:
So far, REST and inmemory catalog follows one pattern (NoSuchTableException) and all other catalogs follows one pattern (AlreadyExistsException).

I tried making Nessie to follow as REST catalog, but it breaks other testcases. I will check more.

Reply:
`replaceTableViaTransactionThatAlreadyExistsAsView`
	`NessieTableOperations.doRefresh() `--> throws `AlreadyExistsException`. But expecting `NoSuchViewException`.

If I fix it, another test case (createOrReplaceTableViaTransactionThatAlreadyExistsAsView) fails.
Because from the same place `doRefresh()` we are expecting two different kind of exceptions. I think test case need to be modified instead of unifying code.

Reply:
Can we follow the in-memory catalog pattern in Nessie?

Reply:
I actually think that the way Nessie currently handles this error case is correct by saying that a view with the same name already exists when you try to create a table. To achieve the same for REST is actually slightly more difficult. There's also a TODO in the test a few lines above where I wanted to improve the error reporting for these 2 particular cases that were adjusted in this test. That being said, I'm +1 on these 2 changes in the test

Reply:
this should be addressed by https://github.com/apache/iceberg/pull/9012/commits/c63be6534a0cfc595905056309b469c6561d414b

Reply:
@nastra: Awesome. Thanks for fixing it for REST and in-memory catalog. That should help in having a unified behaviour.